src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/not.cache.private.cache.with.s-maxage/accept/server.rpt

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2016-2017 The Reaktivity Project
+#
+# The Reaktivity Project licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property newProxyAcceptRef ${nukleus:newRouteRef()} # external scope
+property date ${http_cache:date()} # shared property
+property date2 ${http_cache:datePlus(2)} # shared property
+
+accept await ROUTED_PROXY
+        "nukleus://http-cache/streams/source"
+        option nukleus:route ${newProxyAcceptRef}
+        option nukleus:window 8192
+        option nukleus:transmission "half-duplex"
+accepted
+
+read nukleus:begin.ext ${http:header(":scheme", "http")}
+                       ${http:header(":method", "GET")}
+                       ${http:header(":path", "/path")}
+                       ${http:header(":authority", "localhost:8080")}
+
+
+connected
+
+read closed
+
+write nukleus:begin.ext ${http:header(":status", "200")}
+                        ${http:header("content-length", "11")}
+                        ${http:header("cache-control", "s-maxage=5")}
+                        ${http:header("date", date)}
+                        ${http:header("last-modified", date)}
+                        ${http:header("cache-control", "private")}
+write "hello world"
+
+write close
+
+accepted
+
+read nukleus:begin.ext ${http:header(":scheme", "http")}
+                       ${http:header(":method", "GET")}
+                       ${http:header(":path", "/path")}
+                       ${http:header(":authority", "localhost:8080")}
+
+connected
+
+read closed
+
+write nukleus:begin.ext ${http:header(":status", "200")}
+                        ${http:header("content-length", "11")}
+                        ${http:header("cache-control", "s-maxage=5")}
+                        ${http:header("date", date2)}
```

Comment:
I believe the cache may always add an etag to a cached response.  The etag will be generated if it is not on the response from the origin server.  Consider adding etags to all tests with cached response

Reply:
I believe below code is response for adding etag.

```
      if (!responseHeadersRO.anyMatch(HAS_CACHE_CONTROL))
        {
            builder.item(header -> header.name("cache-control").value("stale-while-revalidate=" + staleWhileRevalidate));
        }
        if (!responseHeadersRO.anyMatch(h -> ETAG.equals(h.name().asString())))
        {
            builder.item(header -> header.name(ETAG).value(etag));
        }
```
When I don't include etag int the response even with missing etag on the client.rpt the test passes. Am I missing something? I thought test should fail or k3po validation works this way

Reply:
And what is reason behind adding one. If one doesn't exist. Just would like to clarify.

Reply:
The extensions on K3po only adds what you specify.  As the etag is typically added at the end of the headers you aren't specifying it so if added it does not match present vs absent.

Reply:
As such you are correct it is not required